bigbone/src/main/kotlin/social/bigbone/api/entity/Notification.kt

```diff
@@ -54,16 +54,41 @@ data class Notification(
      */
     @Serializable
     enum class NotificationType {
+
+        @SerialName("admin.report")
+        ADMIN_REPORT {
+            override val apiName: String = "admin.report"
+        },
+
+        @SerialName("admin.sign_up")
+        ADMIN_SIGN_UP {
+            override val apiName: String = "admin.sign_up"
+        },
+
+        @SerialName("favourite")
+        FAVOURITE,
+
+        @SerialName("follow")
+        FOLLOW,
+
+        @SerialName("follow_request")
+        FOLLOW_REQUEST,
+
         @SerialName("mention")
         MENTION,
 
+        @SerialName("poll")
+        POLL,
+
         @SerialName("reblog")
         REBLOG,
 
-        @SerialName("favourite")
-        FAVOURITE,
+        @SerialName("status")
+        STATUS,
 
-        @SerialName("follow")
-        FOLLOW
+        @SerialName("update")
+        UPDATE;
+
+        open val apiName: String = name.lowercase()
```

Comment:
I had to add `apiName` now after all… 😞 In #310 we thought that we’d only need the api names for tests and decided it’s not worth it to keep the `apiName` parameter up-to-date for that.

But we also need the `NotificationType` for the query parameters `types` and `exclude_types` of the “get all notifications” method. `admin.report` and `admin.sign_up` both contain dots which are not allowed in enum names.

For me, this was the best solution here. If you have a better idea, feel free to comment.

Reply:
Ah, wait. Found a better way! 🥳 

https://github.com/Kotlin/kotlinx.serialization/issues/1482#issuecomment-841308774

Reply:
I did not update other classes as that would be out of scope for this PR.

If you think this is a good way forward, I’m happy to create a ticket.